src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt

```diff
@@ -333,6 +338,13 @@ class PrisonerIepLevelReviewService(
     }
   }
 
+  private suspend fun updateIepLevelWithCurrentFlagToFalse(bookingId: Long) {
+    prisonerIepLevelRepository.findFirstByBookingIdAndCurrentIsTrueOrderByReviewTimeDesc(bookingId)
```

Comment:
Bear in mind at the moment (and potentially in the future unless this is enforced at DB level) more than one record for a given prisoner/`bookingId` could have `current=true`.

So maybe this should really get the list of rows with `bookind=$bookingId AND current=true` and loop through them to set current to false on all the rows.

Reply:
(so basically replace `.findFirstByBookingIdAndCurrentIsTrue...()` with something like `.findAllByBookingIdAndCurrentIsTrue...()` and the `?.let {}` with some sort of loop that set the current field to `false` on all of them...not the most elegant I know)

Reply:
Great spot - addressed